src/rhsm/utils.py

```diff
@@ -276,3 +278,51 @@ def wrapper(*args, **kwargs):
             sys.stderr = stderr
             devnull.close()
     return wrapper
+
+
+class StatusMessage:
+    """Class for temorary reporting.
+
+    While you can call 'print()' and 'clean()' methods directly, the easier
+    way is to use context manager: 'with StatusMessage("Doing stuff..."):'.
+
+    This object will print the description to stdout when called. When the
+    context manager exists (either because the code finished or because an
+    error got raised inside), printed output will be cleared before the
+    program continues with its execution. This ensures that the message will
+    disappear after it's no longer valid.
+    """
+    def __init__(self, description: Optional[str]):
+        if description is None:
+            description = "Transmitting data"
+        self.raw_text = description
+
+        CURSIVE = "\033[3m"
+        RESET = "\033[0m"
+
+        self.text = f"{CURSIVE}{self.raw_text}{RESET}"
+
+        self.quiet = False
+        config = rhsm.config.get_config_parser()
+        if config.get("rhsm", "quiet") == "1":
+            self.quiet = True
+        if not sys.stdout.isatty():
+            self.quiet = True
+
+    def print(self):
+        if self.quiet:
+            return
+        print(self.text, end="\r")
```

Comment:
I would modify this `print()` to something like: `print(f"{self.text}...", end="\r")` to emphasize that the operation is in progress.

Reply:
Not sure if it could be possible to extend it more and display some "waiting animation". Something like this:

https://stackoverflow.com/questions/7039114/waiting-animation-in-command-prompt-python

Reply:
I'm not sure if we can do both the request and the waiting animation. Maybe by opening async loop? I'll try some things.

Reply:
I've added configurable spinning animation (see the new commit). Let me know what you think -- now that the spinner is present, we may not need the `...` dots after the message, it's clear that something is happening.

Reply:
Well done. I would keep dots.